app/src/test/java/com/kks/nimblesurveyjetpackcompose/viewmodel/splash/SplashViewModelTest.kt

```diff
@@ -0,0 +1,42 @@
+package com.kks.nimblesurveyjetpackcompose.viewmodel.splash
+
+import com.kks.nimblesurveyjetpackcompose.base.BaseViewModelTest
+import org.junit.Assert.assertEquals
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class SplashViewModelTest : BaseViewModelTest() {
+
+    private lateinit var viewModel: SplashViewModel
+
+    override fun setup() {
+        super.setup()
+        viewModel = SplashViewModel(dispatcher = testDispatcher)
+    }
+
+    @Test
+    fun `When splash screen is displayed, shouldNavigateToLogin value is false at first`() {
+        val actual = viewModel.shouldNavigateToLogin.value
+        assertEquals(false, actual)
+    }
+
+    @Test
+    fun `When splash screen is displayed, shouldNavigateToLogin value is true after 2 seconds`() =
+        runTest {
+            viewModel.startTimerToNavigateToLogin()
+
+            var actual = false
+            launch {
+                delay(SPLASH_TIME)
+                actual = viewModel.shouldNavigateToLogin.value
+            }
+            advanceUntilIdle()
+
+            assertEquals(true, actual)
```

Comment:
Do we really need to declare a `var actual` for this? Why not directly access `shouldNavigateToLogin`? 🤔 
```suggestion
            assertEquals(true, viewModel.shouldNavigateToLogin.value)
```

Reply:
@ryan-conway I am planning to make the transition animation after I have implemented the `Login UI`. BTW how did you implement that transition? I saw that Compose haven't officially support that animation yet. I saw [this experimental library](https://github.com/mxalbert1996/compose-shared-elements) though.

Reply:
You could combine `Splash` and `Login` into one screen and animate the composables inside it 😉

Reply:
I see. I also had that thought but like you said I need to combine two screens and show and hide the other components.